articles/openshift/howto-deploy-java-eap-app.md

```diff
@@ -0,0 +1,384 @@
+
+
+## Run the application locally
+
+Before deploying the application on OpenShift, we are going to run it locally to verify how it works. For the database, we are going to use a containerized Microsoft SQL Server running on Docker.
+
+### Run the Microsoft SQL database on Docker
+
+Follow the next steps to get the database server running on Docker and configured for the demo application:
+
+1. Start a Docker container running the Microsoft SQL Server. You can get more information about this procedure by checking [this](https://docs.microsoft.com/en-us/sql/linux/quickstart-install-connect-docker) guide.
+
+    ```bash
+    $ sudo docker run \
+    -e 'ACCEPT_EULA=Y' \
+    -e 'SA_PASSWORD=Passw0rd!' \
+    -p 1433:1433 --name mssqlserver -h mssqlserver \
+    -d mcr.microsoft.com/mssql/server:2019-latest
+    ```
+
+1. Login into the server and create the `todos_db` database. This database name is expected by the demo application.
+
+    ```bash
+    $ sudo docker exec -it mssqlserver "bash"
+    mssql@mssqlserver:/$ /opt/mssql-tools/bin/sqlcmd -S localhost -U SA -P 'Passw0rd!'
+    1> CREATE DATABASE todos_db
+    2> GO
+    1> exit
+    mssql@mssqlserver:/$ exit
+    ```
+
+### Run the demo application locally
+
+Follow the next steps to build and run the application locally.
+
+1. Build the Bootable JAR. When we are building the Bootable JAR, we need to specify the database driver version we want to use:
+ 
+    ```bash
+    todo-list (bootable-jar) $ MSSQLSERVER_DRIVER_VERSION=7.4.1.jre11 \
+    mvn clean package
+    ```
+
+1. Launch the Bootable JAR by using the following command. When we are launching the application, we need to pass the required environment variables to configure the data source:
+
+    ```bash
+    todo-list (bootable-jar) $ MSSQLSERVER_USER=SA \
+    MSSQLSERVER_PASSWORD=Passw0rd! \
+    MSSQLSERVER_JNDI=java:/comp/env/jdbc/mssqlds \
+    MSSQLSERVER_DATABASE=todos_db \
+    MSSQLSERVER_HOST=localhost \
+    MSSQLSERVER_PORT=1433 \
+    java -jar ./target/todo-list-bootable.jar -Djboss.node.name=node1
+    ```
+
+1. (Optional) If you want to verify the clustering capabilities, you can also launch more instances of the same application by changing the `jboss.node.name` argument and, to avoid conflicts with the port numbers, shifting the port numbers by using `jboss.socket.binding.port-offset` argument. For example, to launch a second instance that will represent a new pod on OpenShift, you can execute the following command in a new terminal window:
+    
+    ```bash  
+    todo-list (bootable-jar) $ MSSQLSERVER_USER=SA \
+    MSSQLSERVER_PASSWORD=Passw0rd! \
+    MSSQLSERVER_JNDI=java:/comp/env/jdbc/mssqlds \
+    MSSQLSERVER_DATABASE=todos_db \
+    MSSQLSERVER_HOST=localhost \
+    MSSQLSERVER_PORT=1433 \
+    java -jar ./target/todo-list-bootable.jar -Djboss.node.name=node2 -Djboss.socket.binding.port-offset=10
```

Comment:
You use an offset of 10 here but in the example below you use 9080 for the http port, so an offset of 1000.